cms-contentful/src/main/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```diff
@@ -1,17 +1,95 @@
 package io.commercetools.sunrise.cms.contentful;
 
+import com.contentful.java.cda.CDAArray;
+import com.contentful.java.cda.CDAAsset;
+import com.contentful.java.cda.CDAClient;
+import com.contentful.java.cda.CDAEntry;
 import io.commercetools.sunrise.cms.CmsIdentifier;
 import io.commercetools.sunrise.cms.CmsService;
 
+import javax.annotation.Nullable;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 
 public class ContentfulCmsService implements CmsService {
 
+    private static final String ENTRY_TYPE = "content_type";
+    private static final String ENTRY_KEY = "fields.name[match]";
+    private static final String LIMIT = "limit";
+    private static final String ONE = "1";
+    private final CDAClient client;
+
+    ContentfulCmsService(final CDAClient client) {
+        this.client = client;
+    }
+
     @Override
     public CompletionStage<Optional<String>> get(final List<Locale> locales, final CmsIdentifier cmsIdentifier) {
-        return null; // TODO
+        return CompletableFuture.supplyAsync(() -> getEntry(cmsIdentifier)
+                .map(entry -> getLocalizedField(locales, entry, cmsIdentifier.getFieldName()))
+                .orElseGet(Optional::empty));
+    }
+
+    // TODO create models
+    private Optional<CDAEntry> getEntry(final CmsIdentifier cmsIdentifier) {
+        final CDAArray cdaArray = client.fetch(CDAEntry.class)
+                .where(ENTRY_TYPE, cmsIdentifier.getEntryType())
+                .where(ENTRY_KEY, cmsIdentifier.getEntryKey())
+                .where(LIMIT, ONE)
+                .all();
```

Comment:
The CMS service interface is designed to provide an asynchronous API and contentful supports this.

This is a **guess** how the non-blocking contentenful API works:

```
    private CompletionStage<CDAResource> fetch(final CmsIdentifier cmsIdentifier) {
        final CompletableFuture<CDAResource> future = new CompletableFuture<>();
        final CDACallback<CDAArray> callback = new CDACallback(){

            @Override
            protected void onSuccess(final CDAResource result) {
                future.complete(result);
            }

            @Override
            protected void onFailure(final Throwable error) {
                super.onFailure(error);
                future.completeExceptionally(error);
            }
        };
        client.fetch(CDAEntry.class)
                .where(ENTRY_TYPE, cmsIdentifier.getEntryType())
                .where(ENTRY_KEY, cmsIdentifier.getEntryKey())
                .where(LIMIT, ONE)
                .all(callback);
        return future;//will be returned as CompletionStage
    }
```